pyface/font.py

```diff
@@ -0,0 +1,313 @@
+        info += (
+            " or a string with a float value from 50 to 200 followed by '%'"
+        )
+        return info
+
+
+class FontSize(BaseCFloat):
+    """ Trait type for font sizes.
+
+    The is a CFloat trait which also allows values which are keys of the
+    size dictionary, and also ignores trailing 'pt' ot 'px' annotation in
+    string values.  The value stored is a float.
+    """
+
+    #: The default value for the trait.
+    default_value = 12.0
+
+    def __init__(self, default_value=NoDefaultSpecified, **metadata):
+        if default_value != NoDefaultSpecified:
+            default_value = self.validate(None, None, default_value)
+        super().__init__(default_value, **metadata)
+
+    def validate(self, object, name, value):
+        if isinstance(value, str) and (
+                    value.endswith('pt') or value.endswith('px')):
+            value = value[:-2]
+        value = SIZES.get(value, value)
+        value = super().validate(object, name, value)
+        if value <= 0:
+            self.error(object, name, value)
+        return value
+
+    def info(self):
+        info = (
+            "a positive float, a value that can convert to a positive float, "
+        )
+        info += ', '.join(repr(key) for key in SIZES)
+        info += (
+            " or a string with a positive float value followed by 'pt' or 'px'"
+        )
+        return info
+
+
+class Font(HasStrictTraits):
+    """ A toolkit-independent font specification. """
+
+    #: The preferred font families.
+    family = FontFamily()
+
+    #: The weight of the font.
+    weight = FontWeight()
+
+    #: How much the font is expanded or compressed.
+    stretch = FontStretch()
+
+    #: The style of the font.
+    style = FontStyle()
+
+    #: The size of the font.
+    size = FontSize()
+
+    #: The font variants.
+    variants = FontVariants()
+
+    @classmethod
+    def from_toolkit(cls, toolkit_font):
+        """ Create a Font from a toolkit font object.
+
+        Parameters
+        ----------
+        toolkit_font : any
+            A toolkit font to be converted to a corresponding class instance,
+            within the limitations of the options supported by the class.
+        """
+        from pyface.toolkit import toolkit_object
+        toolkit_font_to_properties = toolkit_object(
+            'font:toolkit_font_to_properties')
+
+        return cls(**toolkit_font_to_properties(toolkit_font))
+
+    def to_toolkit(self):
+        """ Create a toolkit font object from the Font instance.
+
+        Returns
+        -------
+        toolkit_font : any
+            A toolkit font which matches the property of the font as
+            closely as possible given the constraints of the toolkit.
+        """
+        from pyface.toolkit import toolkit_object
+        font_to_toolkit_font = toolkit_object('font:font_to_toolkit_font')
+
+        return font_to_toolkit_font(self)
+
+    def __str__(self):
```

Comment:
Can we add a doc string here documenting a few examples of the str reprs of Font traits? It's pretty hard to imagine what the str would look like given the conditionals (the test helped to visualize the complexity)

Reply:
I have added a brief docstring to indicate that it it is CSS style.

Again, this would be clearer with the CSS parsing code available.